File: src/main/java/com/example/demo/entities/LigneBudgetaire.java
package com.example.demo.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;
@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class LigneBudgetaire implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private int version;
    private Double montant;
    private Double montantExecute;
    private Double montantProposeBureau;
    private Double montantProposeDrp;
    @JsonIgnore
    @ManyToOne
    @JoinColumn(name ="idBudget", nullable = false, referencedColumnName = "id")
    private Budget budget;
    @ManyToOne
    @JoinColumn(name ="idCompteBudget", nullable = false, referencedColumnName = "id")
    private CompteBudget compteBudget;
}

File: src/app/demo/model/ligneBudgetaire.ts
import { Budget } from "./budget";
import { CompteBudget } from "./compteBudget";

export class LigneBudgetaire {
	 id?: number;
	 montant?: number;
	 montantProposeBureau?: number;
	 montantProposeDrp?: number;
	 montantExecute?: number;
	 budget?: Budget;
	 compteBudget?: CompteBudget;
	 version =1;
} 